.changeset/many-dancers-fold.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Move `image()` to come from `schema` instead to fix it not working with refine and inside complex types
```

Comment:
Could had a migration section here to show necessary code changes. [Here's an example](https://github.com/withastro/astro/pull/6653/files#diff-b20a481d3c7673a1da8222b0c6428b3b6fac2d1e97316aa4a59efdc122be21adR8) from the Markdoc 0.1 change